driver/manager.go

```diff
@@ -178,3 +181,34 @@ func (d *DriverHandle) HistoryAPISupported(ctx context.Context) bool {
 	})
 	return d.historyAPISupported
 }
+
+func (d *DriverHandle) HostGatewayIP(ctx context.Context) (net.IP, error) {
+	d.hostGatewayIPOnce.Do(func() {
+		if !d.Driver.IsMobyDriver() {
+			return
+		}
+		c, err := d.Client(ctx)
+		if err != nil {
+			d.hostGatewayIPErr = err
+			return
+		}
+		workers, err := c.ListWorkers(ctx)
+		if err != nil {
+			d.hostGatewayIPErr = err
+			return
+		}
+		if len(workers) == 0 {
+			return
```

Comment:
Wondering the same here; would this be an error condition? What does the rest of the build do if there's no workers? (somewhat assuming that would also produce an error somewhere).

Somewhat unrelated question: can this code be used in other situationsions than the `buildx` _CLI_ ? If this code is used in a long lived process, could the state of the world change (as in: `ListWorkers()` later now having actual workers).

Because if that's the case, I don't think we can use a `sync.Once()` for this code ⚠️

Reply:
Workers are spawned on startup when instantiating a new controller such as https://github.com/moby/moby/blob/8d070e30f5a7e8aaa3dd3dd431f98b8c8909fbe2/builder/builder-next/controller.go#L114 and BuildKit would fail to start if there is none available afaik: https://github.com/moby/buildkit/blob/23d38a7220bff812b8a1de07c1f02b4e46cacd7c/cmd/buildkitd/main.go#L752-L754

We also don't support multi workers when solving a request atm iirc: https://github.com/moby/buildkit/blob/23d38a7220bff812b8a1de07c1f02b4e46cacd7c/control/control.go#L326

I changed the logic for this case and takes the very first one that has the label set.